File: 1886-minimum-limit-of-balls-in-a-bag/minimum-limit-of-balls-in-a-bag.py
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        low, high = 1, max(nums)
        
        while low < high:
            mid = (low + high) // 2
            if sum((num - 1) // mid for num in nums) <= maxOperations:
                high = mid
            else:
                low = mid + 1
        
        return high

File: 1886-minimum-limit-of-balls-in-a-bag/minimum-limit-of-balls-in-a-bag.js
/**
 * @param {number[]} nums
 * @param {number} maxOperations
 * @return {number}
 */
var minimumSize = function (nums, maxOperations) {
    let low = 1, high = Math.max(...nums);

    while (low < high) {
        const mid = Math.floor((low + high) / 2);
        let count = 0;

        if (nums.reduce((acc, num) => acc + Math.floor((num - 1) / mid), 0) <= maxOperations) {
            high = mid;
        }
        else {
            low = mid + 1;
        }
    }

    return high;
};